React/Fabric/RCTSurfacePresenter.mm

```diff
@@ -331,7 +331,12 @@ - (void)mountingManager:(RCTMountingManager *)mountingManager willMountComponent
 {
   RCTAssertMainQueue();
 
-  for (id<RCTSurfacePresenterObserver> observer in _observers) {
+  NSArray<id<RCTSurfacePresenterObserver>> *observers;
+  {
+    std::shared_lock<better::shared_mutex> lock(_observerListMutex);
+    observers = [_observers copy];
+  }
+  for (id<RCTSurfacePresenterObserver> observer in observers) {
```

Comment:
Why not iterate over _observers while the lock is acquired?

Reply:
One thing to watch out for is deadlock if any of the observer delegate methods can add/remove observers unless better::shared_mutex is like recursive_mutex.

Reply:
@shergin What I concerned is the same as @ide , we have no control what delegate would do. But if this deadlock situation shouldn't happened. I can remove copy.

Reply:
`willMountComponentsWithRootTag:` is being called quite often and perf sensitive, so please do.

Reply:
Done!